tools/clang/test/CodeGenSPIRV/cf.return.void.hlsl

```diff
@@ -0,0 +1,9 @@
+// RUN: %dxc -T ps_6_0 -E main
+
+void A() {
+}
+
+void main() {
+  // CHECK: OpReturn
+  return A();
+}
```

Comment:
Did we double check with the DXIL developers if this should be allowed?  In C++ this is considered a syntax error, so I would have expected the same for HLSL.  I guess they are following C.  From the bug, it seems like it is being used.

Reply:
DXIL does allow it, but I don't know if it _should_.

Reply:
> Did we double check with the DXIL developers if this should be allowed? In C++ this is considered a syntax error, so I would have expected the same for HLSL. I guess they are following C. From the bug, it seems like it is being used.

No, in C++ it's allowed. "In a function returning (possibly cv-qualified) void, the return statement with expression can be used, if the expression type is (possibly cv-qualified) void. "